components/service/pocket/src/main/java/mozilla/components/service/pocket/stories/PocketStoriesUseCases.kt

```diff
@@ -10,101 +10,67 @@ import mozilla.components.concept.fetch.Client
+object PocketStoriesUseCases {
+    @VisibleForTesting internal var fetchClient: Client? = null
 
     /**
-     * Allows for refreshing the list of pocket stories we have cached.
-     *
-     * @param context Android Context. Prefer sending application context to limit the possibility of even small leaks.
-     * @param pocketToken Pocket OAuth request token used for downloading recommended stories.
-     * @param pocketItemsCount How many recommended stories to download.
-     *     Once downloaded these stories will replace any that we already had cached.
-     * @param pocketItemsLocale The ISO-639 locale for Pocket recommended stories.
+     * Returns the current locally persisted list of Pocket recommended stories.
      */
-    internal inner class RefreshPocketStories(
-        @VisibleForTesting
-        internal val context: Context,
-        @VisibleForTesting
-        internal val pocketItemsCount: Int,
-        @VisibleForTesting
-        internal val pocketItemsLocale: String
-    ) {
-        /**
-         * Do a full download from Pocket -> persist locally cycle for recommended stories.
-         */
-        suspend operator fun invoke(): Boolean {
-            val apiKey = pocketApiKey
-            val client = fetchClient
-            if (apiKey == null || client == null) {
-                Log.log(
-                    Log.Priority.ERROR,
-                    this.javaClass.simpleName,
-                    message = "Cannot download new stories. Service has incomplete setup"
-                )
-                return false
-            }
-
-            val pocket = getPocketEndpoint(apiKey, client)
-            val response = pocket.getTopStories(pocketItemsCount, pocketItemsLocale)
-
-            if (response is PocketResponse.Success) {
-                getPocketRepository(context)
-                    .addAllPocketPocketRecommendedStories(response.data)
-                return true
-            }
-
-            return false
-        }
+    suspend fun getPocketStories(context: Context): List<PocketRecommendedStory> {
+        return PocketRecommendationsRepository.getPocketRecommendedStories(context)
     }
 
     /**
-     * Allows for querying the list of locally available Pocket recommended stories.
+     * Convenience method for setting the Pocket api key and the HTTP Client which will be used
+     * for all REST communication with the Pocket server.
+     *
+     * Already downloaded data can still be queried but no new data can be downloaded until this data is set.
      */
-    inner class GetPocketStories(private val context: Context) {
-        /**
-         * Returns the current locally persisted list of Pocket recommended stories.
-         */
-        suspend operator fun invoke(): List<PocketRecommendedStory> {
-            return getPocketRepository(context)
-                .getPocketRecommendedStories()
-        }
+    internal fun initialize(client: Client) {
+        this.fetchClient = client
     }
```

Comment:
This is now kind of a fake constructor, so I'm still not sure why not construct it with the dependency?

Reply:
Reverted to use the same previous implementation.